packages/remirror__extension-list/src/task-list-extension.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  ApplySchemaAttributes,
+  assertGet,
+  command,
+  CommandFunction,
+  ExtensionPriority,
+  ExtensionTag,
+  getMatchString,
+  InputRule,
+  keyBinding,
+  KeyBindingProps,
+  NamedShortcut,
+  NodeExtension,
+  NodeExtensionSpec,
+  nodeInputRule,
+  NodeSpecOverride,
+  NodeViewMethod,
+  ProsemirrorNode,
+} from '@remirror/core';
+import { ExtensionListMessages as Messages } from '@remirror/messages';
+import { NodeSelection, TextSelection } from '@remirror/pm/state';
+import { ExtensionListTheme } from '@remirror/theme';
+
+import { toggleList } from './list-commands';
+import { TaskListItemExtension } from './task-list-item-extension';
+
+/**
+ * Create the node for a task list.
+ */
+export class TaskListExtension extends NodeExtension {
+  get name() {
+    return 'taskList' as const;
+  }
+
+  createTags() {
+    return [ExtensionTag.Block, ExtensionTag.ListContainerNode];
+  }
+
+  createNodeSpec(extra: ApplySchemaAttributes, override: NodeSpecOverride): NodeExtensionSpec {
+    return {
+      content: 'taskListItem+',
+      ...override,
+      attrs: extra.defaults(),
+      parseDOM: [{ tag: 'ul', getAttrs: extra.parse }, ...(override.parseDOM ?? [])],
+      toDOM: (node) => ['ul', extra.dom(node), 0],
```

Comment:
I think your concern and solution are very reasonable. Let me write some test cases to make sure it works as expected.

Reply:
Fixed it. I don't know why but it seems that I need not only add `data-task-list` to the list elements but also add `data-task-list-item` to the list item elements so that `TaskListItemExtension`'s `parseDOM` will also work.